test/util/helpers.ts

```diff
@@ -33,10 +28,7 @@ export interface ProviderRequestStub<
   Result extends Json,
 > {
   request: Partial<JsonRpcRequest<Params>>;
-  response: (
-    request: JsonRpcRequest<Params>,
-    callNumber: number,
-  ) => JsonRpcResponse<Result>;
+  response: (callNumber: number) => Promise<Result>;
```

Comment:
Since this is no longer expected to be the response object, do we want to call this `result` instead?